src/FormElement/ScheduleElement.php

```diff
@@ -0,0 +1,592 @@
+
+                            $rule = new RRule("FREQ=MONTHLY;INTERVAL=$interval;BYMONTHDAY=$byMonth");
+
+                            break;
+                        }
+                    // Fall-through to the next switch case
+                    case RRule::YEARLY:
+                        $rule = "FREQ=MONTHLY;INTERVAL=$interval;";
+                        if ($customFrequency === RRule::YEARLY) {
+                            $runsOn = $this->annuallyFields->getValue('runsOnThe', 'n');
+                            $month = $this->annuallyFields->getValue('month', (int) $this->start->format('m'));
+                            if (is_string($month)) {
+                                $datetime = DateTime::createFromFormat('!M', $month);
+                                if (! $datetime) {
+                                    throw new InvalidArgumentException(sprintf('Invalid month provided: %s', $month));
+                                }
+
+                                $month = (int) $datetime->format('m');
+                            }
+
+                            $rule = "FREQ=YEARLY;INTERVAL=1;BYMONTH=$month;";
+                            if ($runsOn === 'n') {
+                                $rule = new RRule($rule);
+
+                                break;
+                            }
+                        }
+
+                        $element = $this->monthlyFields;
+                        if ($customFrequency === RRule::YEARLY) {
+                            $element = $this->annuallyFields;
+                        }
+
+                        $runDay = $element->getValue('day', $element::$everyDay);
+                        $ordinal = $element->getValue('ordinal', $element::$first);
+                        $position = $element->getOrdinalAsInteger($ordinal);
+
+                        if ($runDay === $element::$everyDay) {
+                            $rule .= "BYDAY=MO,TU,WE,TH,FR,SA,SU;BYSETPOS=$position";
+                        } elseif ($runDay === $element::$everyWeekday) {
+                            $rule .= "BYDAY=MO,TU,WE,TH,FR;BYSETPOS=$position";
+                        } elseif ($runDay === $element::$everyWeekend) {
+                            $rule .= "BYDAY=SA,SU;BYSETPOS=$position";
+                        } else {
+                            $rule .= sprintf('BYDAY=%d%s', $position, $runDay);
+                        }
+
+                        $rule = new RRule($rule);
+
+                        break;
+                    default:
+                        throw new LogicException(sprintf('Custom frequency %s is not supported!', $customFrequency));
+                }
+        }
+
+        $rule->startAt($start);
+        if (parent::getValue('use-end-time', 'n') === 'y') {
+            $rule->endAt(parent::getValue('end'));
+        }
+
+        return $rule;
+    }
+
+    public function setValue($value)
+    {
+        $rule = $value['frequency'] ?? null;
```

Comment:
If `$value` is not what has been submitted by the browser, it can only be the rule itself already. There's no key `'frequency'` if I populate the element manually.